csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -156,17 +157,22 @@ public string Format(IMessage message)
         /// <param name="writer">The TextWriter to write the formatted message to.</param>
         /// <returns>The formatted message.</returns>
         public void Format(IMessage message, TextWriter writer)
+        {
+            FormatInternal(message, writer, depth: 0);
+        }
+
+        private void FormatInternal(IMessage message, TextWriter writer, int depth)
```

Comment:
To calculate indentation, `Format` needs to know the current depth.
I added a _private_ overload to make sure the public API is not changed.

However, it would be useful to make it public, to allow embedding the json inside another json string (which is what I need personally). If the reviewer sees no objection I would prefer to add a public overload.

Reply:
Made the overload public, with `<remark>` explaining the delegation.
Renamed `depth` to `indentationLevel` for clarity